lib/host.py

```diff
@@ -97,13 +97,15 @@ def create_file(self, filename, text):
             file.flush()
             self.scp(file.name, filename)
 
-    def add_xcpng_repo(self, name):
+    def add_xcpng_repo(self, name, base_repo='xcp-ng'):
+        assert base_repo in ['xcp-ng', 'vates']
+        base_repo_url = 'http://mirrors.xcp-ng.org/' if base_repo == 'xcp-ng' else 'https://repo.vates.tech/xcp-ng/'
```

Comment:
There's an assert, so no typo will select the fallback. We can rework the code if/when we add a third component.

Reply:
> There's an assert, so no typo will select the fallback. We can rework the code if/when we add a third component.

The idiom with assert makes it so that things have to be changed in 2 places; the dict idiom only one single place.
And I think it's much more visual.

> Regarding repo_id

`repo_nickname` would be more on the point?

Reply:
> repo_nickname would be more on the point?

It's really a "base" something. The thing is repo may have several meanings. Either a location on a mirror for a whole distro, or a specific yum repository such as https://updates.xcp-ng.org/8/8.3/lab/x86_64/

There must be a better name than `base_repo`, but here it's just good enough and the other suggestions don't look better to me. Let's keep `base_repo`.

Reply:
> > There's an assert, so no typo will select the fallback. We can rework the code if/when we add a third component.
> 
> The idiom with assert makes it so that things have to be changed in 2 places; the dict idiom only one single place. And I think it's much more visual.

This is true, but if we change it now, we already consume one change so you're back to two. And it's definitely not like we're going to have trouble changing it the way it is now if we need to.